tools/test_api.py

```diff
@@ -2159,7 +2161,9 @@ def build_tests(tests, base_source_paths, build_path, target, toolchain_name,
             'verbose': verbose,
             'app_config': app_config,
             'build_profile': build_profile,
-            'silent': True
+            'silent': True,
+            # Coverage requires main(). So if enabled on any module, enable it on test as well.
+            'coverage_filter': ".*" if coverage_filter else []
```

Comment:
Just to be clear here, this will enable coverage on all source files contained within the test case folder, correct? So say my test is in the directory `/mbed-os/TESTS/coverage_tests/testcase1/` and inside there I have the following files: `main.cpp`, `test_library.cpp`, and `test_library.h`. Coverage would be enabled for all of those files, not just `main.cpp`, correct? That shouldn't be a problem in most cases, just want to clarify.